lib/galaxy/webapps/galaxy/services/tools.py

```diff
@@ -55,27 +64,40 @@ def __init__(
         self.toolbox_search = toolbox_search
         self.history_manager = history_manager
 
+    def create_temp_file(self, trans, files) -> Dict[str, FilesPayload]:
+        files_payload = {}
+        for i, upload_file in enumerate(files):
+            with tempfile.NamedTemporaryFile(
+                dir=trans.app.config.new_file_path, prefix="upload_file_data_", delete=False
+            ) as dest:
+                shutil.copyfileobj(upload_file.file, dest)  # type: ignore[misc]  # https://github.com/python/mypy/issues/15031
+            upload_file.file.close()
+            files_payload[f"files_{i}|file_data"] = FilesPayload(
+                filename=upload_file.filename, local_filename=dest.name
+            )
+        return files_payload
+
+    def create_temp_file_execute(self, trans, upload_file) -> FilesPayload:
+        with tempfile.NamedTemporaryFile(
```

Comment:
Same here, we know the types of trans and upload_file, it is better to be explicit about the types here.